src/components/common/GeolocationControl.js

```diff
@@ -0,0 +1,117 @@
+import React, { Component } from 'react';
+import FormControl from '@material-ui/core/FormControl';
+import { withStyles } from '@material-ui/core/styles';
+import { withTranslation } from 'react-i18next';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import Switch from '@material-ui/core/Switch';
+import Button from '@material-ui/core/Button';
+import FormControlLabel from '@material-ui/core/FormControlLabel';
+import { getGeolocationEnabled, setGeolocationEnabled } from '../../actions';
+import Loader from './Loader';
+
+const styles = theme => ({
+  formControl: {
+    margin: theme.spacing(),
+    minWidth: 120,
+  },
+});
+
+class GeolocationControl extends Component {
+  static propTypes = {
+    geolocationEnabled: PropTypes.bool.isRequired,
+    activity: PropTypes.bool.isRequired,
+    t: PropTypes.func.isRequired,
+    dispatchGetGeolocationEnabled: PropTypes.func.isRequired,
+    dispatchSetGeolocationEnabled: PropTypes.func.isRequired,
+    classes: PropTypes.shape({
+      formControl: PropTypes.string.isRequired,
+    }).isRequired,
+    isButton: PropTypes.bool,
+  };
+
+  static defaultProps = {
+    isButton: false,
+  };
+
+  constructor(props) {
+    super(props);
+    const { dispatchGetGeolocationEnabled } = this.props;
+    dispatchGetGeolocationEnabled();
+  }
+
+  handleClick = async () => {
+    const { dispatchSetGeolocationEnabled } = this.props;
+    dispatchSetGeolocationEnabled(true);
+  };
+
+  handleChange = async ({ target }) => {
+    const { dispatchSetGeolocationEnabled } = this.props;
+    const { checked } = target;
+    dispatchSetGeolocationEnabled(checked);
+  };
+
+  render() {
+    const {
+      classes,
+      t,
+      geolocationEnabled,
+      activity,
+      isButton = false,
```

Comment:
Can we change this to `button`? I like to keep booleans without the `is`, so that only functions are verbs. Or maybe keep it as an enum `type` with possible values `button` and `switch`. If you do that then in proptypes you can use `oneOf` to designate the possible values. As you prefer.